### PR TITLE
Fee change timelock

### PR DESCRIPTION
**Pull request by robdoesstuff**

### CODE REVIEW

**Review by kevinday — APPROVED** [review]

File: src/PatchworkProtocol.sol [hunk 1]
```diff
@@ -63,12 +63,18 @@ contract PatchworkProtocol is IPatchworkProtocol, Ownable, ReentrancyGuard {
     /// Current protocol fee configuration
     ProtocolFeeConfig private _protocolFeeConfig;
 
+    /// Proposed protocol fee configuration
+    mapping(string => ProposedProtocolFeeConfig) private _proposedProtocolFeeConfigs;
```
Comment (kevinday) [inline src/PatchworkProtocol.sol:67]: naming nit: can we just call these ProposedFeeConfig and _proposedFeeConfigs since it is tracking both protocol and scope fee proposals?

File: src/IPatchworkProtocol.sol [hunk 2]
```diff
@@ -258,6 +268,15 @@ interface IPatchworkProtocol {
         bool active; /// true for present
     }
 
+    /**
```
Comment (kevinday) [inline src/IPatchworkProtocol.sol:271]: Not part of this change, but is it more consistent to define ProtoclFeeOverride as

struct ProtocolFeeOverride {
    ProtocolFeeConfig config;
    bool active;
}

Also mentioned elsewhere, but I think it makes more sense to name these generically as FeeConfig/Override in all places where these are being used for both scope and protocol fees.

File: src/IPatchworkProtocol.sol [hunk 3]
```diff
@@ -715,11 +754,19 @@ interface IPatchworkProtocol {
...
 
     /**
-    @notice Set the protocol fee configuration
+    @notice Proposes a protocol fee configuration
     @dev must be protocol owner or banker to call
+    @dev configuration does not apply until commitProtocolFeeConfig is called
     @param config The protocol fee configuration to be set
     */
-    function setProtocolFeeConfig(ProtocolFeeConfig memory config) external;
+    function proposeProtocolFeeConfig(ProtocolFeeConfig memory config) external;
```
Comment (kevinday) [inline src/IPatchworkProtocol.sol:762]: Is it worth adding getters for proposals? Or just have people rely on logs for that?

File: src/PatchworkProtocol.sol [hunk 4]
```diff
@@ -371,14 +390,45 @@ contract PatchworkProtocol is IPatchworkProtocol, Ownable, ReentrancyGuard {
...
+    function commitScopeFeeOverride(string memory scopeName) public onlyProtoOwnerBanker {
+        (ProtocolFeeConfig memory config, bool active) = _commitFeeOverride(scopeName);
+        ProtocolFeeOverride memory feeOverride = ProtocolFeeOverride(config.mintBp, config.patchBp, config.assignBp, active);
+        if (!active) {
             delete _scopeFeeOverrides[scopeName];
         } else {
-            _scopeFeeOverrides[scopeName] = config;
+            _scopeFeeOverrides[scopeName] = feeOverride;
+        }
+        emit ScopeFeeOverrideCommit(feeOverride);
```
Comment (kevinday) [inline src/PatchworkProtocol.sol:412]: Same comment as above, include scope name in event

File: src/IPatchworkProtocol.sol (hunk 3 above)
Comment (robdoesstuff) [reply src/IPatchworkProtocol.sol]: I was thinking logs are fine

**Review by kevinday — APPROVED** [review]
